Split neutron backend test based on executor dependency

JakeStevens opened the pull request:
Summary:
The current test_neutron_backend has a hard requirement on tflite due to the Executor comparing the output.

Not all systems have tflite, but we would like to have a sanity check on the backend still.

To achieve this, this PR splits the existing test into two files

test_neutron_backend contains the tests that do not use any Executor logic (ie, no end to end comparison of outputs)
test_neutron_backend_executor contains the rest of the tests.

Differential Revision: D77248751

JakeStevens commented:
> In general looks good to me. How do you plan to execute tests only in particular files? I believe pytest will fail to run anything if there's at least single file with problematic import. Only working solutions how to execute this are imho 1) single file's tests execution 2) predefined list of "files with supported tests".
> 
> Please rebase to main to include nxp-unittest workflow.

This is an issue specific to our internal workflow, which does not use the OSS flow but uses BUCK entirely. So, you just only include the files you want in your BUCK target, very easy (basically a special form of your #2 suggestion).

Will rebase

robert-kalmar commented:
LGTM, 
executed the trunk CI to invoke the nxp-unittests too to verify the change.

JakeStevens committed:
Split neutron backend test based on executor dependency (#11934)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/11934

The current test_neutron_backend has a hard requirement on tflite due to the Executor comparing the output.

Not all systems have tflite, but we would like to have a sanity check on the backend still.

To achieve this, this PR splits the existing test into two files

test_neutron_backend contains the tests that do not use any Executor logic (ie, no end to end comparison of outputs)
test_neutron_backend_executor contains the rest of the tests.

Reviewed By: digantdesai

Differential Revision: D77248751
